app/src/main/java/com/example/myapplication/CredentialsManager.kt

```diff
@@ -0,0 +1,50 @@
+package com.example.myapplication
+
+class CredentialsManager {
+
+    private val emailPattern = ("[a-zA-Z0-9\\+\\%\\-\\+]{1,256}" +
+            "\\@" +
+            "[a-zA-Z0-9][0-zA-Z0-9\\-]{0,64}" +
+            "(" +
+            "\\." +
+            "[a-zA-Z0-9][0-zA-Z0-9\\-]{0,25}" +
+            ")+").toRegex()
+
+    fun isValidEmail(email: String): Boolean {
```

Comment:
isEmailValid seems more human for me, but since you keep it consistent - it's your preference